### PR TITLE
Add Malicious Site Protections Configuration

### PR DESCRIPTION
**Pull request by not-a-rootkit**
**Asana Task/Github Issue:** https://app.asana.com/0/72649045549333/1208815326657419/f

## Description
Currently the malicious site protections config is named phishingDetection but as we're adding more data classes to it I'm proposing we rename it to maliciousSiteProtection for consistency.
The config currently contains the following fields:
 - allowErrorPage: when true, the phishing detection feature is enabled and checks are performed for all navigation events.
 - allowPreferencesToggle: when true, there is a settings toggle to allow users to enable/disable the feature.

We've discussed removing the subfeature flags for `allowErrorPage` and `allowPreferencesToggle` so it's combined in just one main flag for the whole feature, and adding settings:

 -  when the main feature flag is enabled, we allow phishing + malware protections to run in the browser
 - settings.hashPrefixUpdateFrequency: currently set to 20 minutes
 - settings.filterSetUpdateFrequency: currently set to 12 hours (in minutes: 720)

### Feature change process:

- [x] I have added a [schema](https://github.com/duckduckgo/privacy-configuration/tree/main/schema) to validate this feature change
- [x] I have tested this change locally
- [x] This code for the config change is ready to merge - yes for Android + macOS.
- [x] This feature was covered by a tech design: https://app.asana.com/0/481882893211075/1207273224076495/f

#### Additional info:

- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] This code for the config change is ready
- [ ] This change was covered by a ship review

#### Reference

-   [Config Reviewer Documentation](https://app.asana.com/0/1200890834746050/1204443212791216/f)
-   [Config Maintainer Documentation](https://app.asana.com/0/1200890834746050/1200573250322769/f)
-   [Feature Implementer Documentation](https://app.asana.com/0/1200890834746050/1201498956177210/f)

### CODE REVIEW

**Comment by not-a-rootkit** [comment]
FYI we're still discussing whether we want this much granular control over the feature. TBD Monday.

File: overrides/ios-override.json [hunk 1]
```diff
@@ -582,16 +582,13 @@
                 }
             }
         },
-        "phishingDetection": {
```
Comment (jonathanKingston) [inline overrides/ios-override.json:585]: This never got rolled out right?

File: features/malicious-site-protection.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "_meta": {
+        "description": "Malicious Site Protection",
```
Comment (jonathanKingston) [inline features/malicious-site-protection.json:3]: Should we explain the rename in the phishing protection feature (aka it's deprecated)

File: index.js [hunk 3]
```diff
@@ -142,6 +142,7 @@ const featuresToIncludeTempUnprotectedExceptions = [
     'nonTracking3pCookies',
     'performanceMetrics',
     'phishingDetection',
+    'maliciousSiteProtection',
```
Comment (jonathanKingston) [inline index.js:145]: I suspect we never wanted that here. This is why there are "exceptions"

File: schema/features/malicious-site-protection.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Feature, CSSInjectFeatureSettings } from '../feature';
+
+type State = 'enabled' | 'disabled' | 'internal';
+type StateObject = {
+    state: State;
+};
```
Comment (jonathanKingston) [inline schema/features/malicious-site-protection.ts:6]:
```suggestion
```

Comment (jonathanKingston) [reply schema/features/malicious-site-protection.ts:6]: (I think this is unused anyway, looks like we don't have a lint that prevents unused types)

File: overrides/ios-override.json (hunk 1 above)
Comment (not-a-rootkit) [reply overrides/ios-override.json:585]: Only internally yes.

File: index.js (hunk 3 above)
Comment (not-a-rootkit) [reply index.js:145]: I thought we wanted exceptions in general, I just want to be able to define them manually though. If we remove it from here can we still do that in the `overrides/*.json`?

Comment (jonathanKingston) [reply index.js]: nah just means temp allowlisted sites get an exception here. It's for website investigations where we don't know the cause.

We can still use overrides as you say :)

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**